clash-cores/src/Clash/Cores/Xilinx/Ethernet/Gmii/Types.hs

```diff
@@ -0,0 +1,181 @@
+{-# LANGUAGE LambdaCase #-}
+{-# OPTIONS_GHC -fconstraint-solver-iterations=10 #-}
+{-# LANGUAGE RecordWildCards #-}
+-- | Contains the types used in the implementation of the GMII to SGMII bridge wrapper.
+-- LogiCORE IP Ethernet 1000BASE-X PCS/PMA or SGMII
+module Clash.Cores.Xilinx.Ethernet.Gmii.Types where
+
+import Clash.Prelude
+
+data Lvds = Lvds{lvdsP :: Bit, lvdsN :: Bit}
+  deriving (Generic, NFDataX, BitPack)
+
+type Speed_10_100 = Bool
+type Speed_100 = Bool
+
+-- TODO: Incorporate in 'Clash.Prelude'?
+-- | Gets time in 'Picoseconds' from time in 'Nanoseconds'.
+type Nanoseconds  (ns :: Nat) = Picoseconds  (1000 * ns)
+-- | Gets time in 'Picoseconds' from time in 'Picoseconds', essentially 'id'.
+type Picoseconds  (ps :: Nat) = ps
+-- | A generic GMII interface
+
+-- TODO: Move to re-usable module?
+-- | Represents a GMII interface.
+data Gmii = Gmii
+ { gmiiData :: BitVector 8
+ , gmiiValid :: Bit
+ , gmiiError :: Bit
+ } deriving (Generic, BitPack, NFDataX)
+
+type SignalDetect = Bool
+type RestartAutoNeg = Bool
+type ConfigValid = Bool
+type BridgeAnConfigValid = Bool
```

Comment:
If these names are important to keep around, then use newtypes instead of type synonyms. Type synonyms are easily lost in type inference and error messages. Otherwise simply drop them and use `Bool` at their current use-sites instead.